packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/dataservice/ModelController.java

```diff
@@ -342,8 +346,24 @@ ResponseEntity<Model> updateModel(
 			// TerariumAsset have a name field, but it's not used for the model name outside
 			// the front-end.
 			final Optional<Model> updated = modelService.updateAsset(model, permission);
-			return updated.map(ResponseEntity::ok)
-					.orElseGet(() -> ResponseEntity.notFound().build());
+
+			if (updated.isEmpty()) {
+				return ResponseEntity.notFound().build();
+			}
+
+			final Model updatedModel = updated.get();
+
+			if (updatedModel.getPublicAsset() && !updatedModel.getTemporary()) {
+				try {
+					final String amr = objectMapper.writeValueAsString(updatedModel);
+					final TerariumAssetEmbeddings embeddings = embeddingService.generateEmbeddings(amr);
+					modelService.uploadEmbeddings(model.getId(), embeddings, permission);
+				} catch (final Exception e) {
+					log.warn("Unable to update embeddings for model " + model.getId(), e);
+				}
+			}
```

Comment:
Note we may have a lot of small updates from the user-interface from things that @shawnyama is doing (updating metadata and sources). We probably want a debounce-mechanism at the UI side of things otherwise this may become expensive.

Reply:
Good to know. I could just disable it and only generate them on model alignment instead?

Reply:
Nah, we should go okay for now. 
More of a anchor/reminder here to chat when Shawn is back.